### PR TITLE
Update #r pacakges tooling RFC

### PR DESCRIPTION
**Pull request by dsyme**
cc @KevinRansom - could you check the whole RFC please?

### CODE REVIEW

**Review by KevinRansom — APPROVED** [review]

File: tooling/FST-1027-fsi-references.md [hunk 1]
```diff
@@ -10,40 +10,23 @@
...
-
-This split for .NET Core necessitates the introduction of `#r "nuget: name, version"` and `#r "paket: paket-command"` to simplify referencing dependencies in F# script files for .NET Core.  The intention is that FSI Scripts in the future will always specify dependencies via `#r "nuget` or `#r "paket` (or a future extended command, like `#r "project"`), allowing only FSI to be concerned with referencing the correct assemblies.
-
-There are two primary considerations here: design-time behavior and runtime behavior.
+.NET Core necessitates the introduction of `#r "nuget: name, version"` and `#r "paket: paket-command"` to simplify referencing dependencies in F# script files for .NET Core.  The intention is that FSI Scripts in the future will always specify dependencies via `#r "nuget` or `#r "paket` (or a future extended command, like `#r "project"`), allowing only FSI to be concerned with referencing the correct assemblies.
```
Comment (cartermp) [inline tooling/FST-1027-fsi-references.md:29]:
```suggestion
.NET Core necessitates the introduction of `#r "nuget: name, version"` to simplify referencing dependencies in F# script files for .NET Core.  The intention is that FSI Scripts in the future will always specify dependencies via `#r "nuget:...` (or other extensions, like `#r "paket:..."`, `#r "frameworkreference:...", `#r "project:..."`, etc.), allowing only FSI to be concerned with referencing the correct assemblies.
```

Paket is not built in and would have to be an extension. Also added some others that have come up as a request

File: tooling/FST-1027-fsi-references.md [hunk 2]
```diff
@@ -145,6 +128,17 @@ This is the interactive scenario.  Interactively in an active FSI session, someo
...
 As above, if they somehow specify something which is already loaded, we should just let FSI attempt to load it and generate the error.
 
+
+### Referencing packages containing native DLLs 
+
+Referencing packages containing native DLLs is supported with some limitations.  
+
+The support is achieved by  adding an event handler to the .NET Core event `AssemblyLoadContext.Default.ResolvingUnmanagedDll`, which is triggered when resolving an unmanaged assembly in the context of a .NET assembly (e.g. a DllImport).   
+
+This handler consults current architecture and platform settings plus resolved package metadata and files across all dynamically referenced packages to look for a matching native DLL and then dynamically loads that DLL using an internal NativeAssemblyLoadContext that implements `LoadNativeLibrary` via `LoadUnmanagedDllFromPath`.
```
Comment (cartermp) [inline tooling/FST-1027-fsi-references.md:138]:
```suggestion
This handler consults current architecture and platform settings plus resolved package metadata and files across all dynamically referenced packages to look for a matching native DLL and then dynamically loads that DLL using an internal `NativeAssemblyLoadContext` that implements `LoadNativeLibrary` via `LoadUnmanagedDllFromPath`.  
```

File: tooling/FST-1027-fsi-references.md [hunk 3]
```diff
@@ -182,13 +176,15 @@ In context of tooling, the location are scanned initially once per interactive s
 
 If a handler key (such as `nuget` or `paket`) is found several times, report a warning showing location of assemblies and showing which one was picked (we apply same order of precedence as for finding the assemblies).
 
+# Tooling
+
+Language service and FCS tooling must be efficient given incremental editing of files.
```
Comment (cartermp) [inline tooling/FST-1027-fsi-references.md:181]: I suppose more detail needs to be given here, perhaps clarifying that design-time resolution of scripts is orthogonal to runtime and carries some concerns:

* Editor performance
* A delay in various tooling due to nuget resolution happening in the background
* Ensuring that we don't re-resolve things unnecessarily
* Future extensions to indicate background work is happening, like how project integration tooling does today

Comment (dsyme) [reply tooling/FST-1027-fsi-references.md:181]: ok

**Pull request merged**